user_model.py

```diff
@@ -105,14 +105,16 @@ def submit(self):
             user_collection.create_index('email', unique=True)
         return_obj = MongoReturn()
         try:
-            result = user_collection.insert_one(self._to_dict())
-            if result.acknowledged and result.inserted_id:
+            result = user_collection.update_one({'email': self.email}, {"$set": self._to_dict()}, upsert=True)
+            if result.acknowledged and result.upserted_id:
                 return_obj.success = True
                 return_obj.count = 1
                 return_obj.message = 'Document inserted successfully'
             else:
-                return_obj.success = False
-                return_obj.message = "Failed to insert document"
+                return_obj.success = True
```

Comment:
This else condition does not seem correct.  if result.acknowledged is false or result.upserted.id is false you are setting success to True. In other words, if there was a problem with the update you are saying "success = True". Actually there is no situation where we ever return success= False.

Mongo manual at https://docs.mongodb.com/manual/reference/method/db.collection.updateOne/